tst/launch_ci_runner.py

```diff
@@ -0,0 +1,239 @@
+        sys.exit(1)
+    return response.json()
+
+
+def update_status(
+    commit_sha, state, description, context="Continuous Integration / darwin_volta-x86"
+):
+    url = f"https://api.github.com/repos/lanl/jaybenne/statuses/{commit_sha}"
+    headers = {"Authorization": f"token {GITHUB_TOKEN}"}
+    data = {"state": state, "description": description, "context": context}
+    response = requests.post(url, headers=headers, data=json.dumps(data))
+    if response.status_code != 201:
+        print(f"Error setting status: {response.status_code}")
+        print(response.text)
+        sys.exit(1)
+
+
+def run_tests_in_temp_dir(pr_number, head_repo, head_ref, output_dir):
+    current_dir = os.getcwd()
+
+    # Create a temporary directory
+    with tempfile.TemporaryDirectory() as temp_dir:
+        print(f"Using temporary directory: {temp_dir}")
+
+        # Clone the repository into the temporary directory
+        subprocess.run(["git", "clone", head_repo, temp_dir], check=True)
+        os.chdir(temp_dir)
+
+        # Checkout the PR branch
+        subprocess.run(["git", "pull", "--no-rebase", "origin", head_ref], check=True)
+
+        # Update submodules
+        subprocess.run(
+            ["git", "submodule", "update", "--init", "--recursive"], check=True
+        )
+
+        # Run the tests
+        os.chdir(os.path.join(temp_dir, "tst"))
+        build_dir = os.path.join(temp_dir, "build")
+
+        # Run subprocess command to compile code and launch run_tests.py
+        test_command = [
+            "bash",
+            "-c",
+            "source ../env/bash && build_jaybenne -b "
+            + build_dir
+            + " -f && cd "
+            + os.path.join(temp_dir, "tst")
+            + " && ./stepdiff.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff.in"
+            + " --use_mpiexec",
+            +" && ./stepdiff.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_ddmc.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_ddmc.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_ddmc.in --use_mpiexec --mpi_nthreads 8",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_hybrid.in --use_mpiexec",
+            +" && ./stepdiff_smr.py --executable "
+            + os.path.join(build_dir, "mcblock")
+            + " --input ../inputs/stepdiff_smr_hybrid.in --use_mpiexec --mpi_nthreads 8",
```

Comment:
We need a `run_tests.py [suite]` script that can automatically launch a number of tests at once for here and the CI workflow file

Reply:
Maybe not following - how would that be different than the above command block?

Reply:
Oh I mean it would just replace that giant command block both here and in the CI (and maybe in users local workflows) with just like `./tst/run_tests.py gpu.suite`